### DISCUSSION TITLE
Bun v1.0.4

### DISCUSSION THREAD

**Discussion by Jarred-Sumner** [oven-sh/bun#6262] [Announcements]
To install Bun v1.0.4:
```bash
curl https://bun.sh/install | bash
# or you can use npm
# npm install -g bun
```

To upgrade to Bun v1.0.4:
```bash
bun upgrade
```

**[Read Bun v1.0.4's release notes on Bun's blog](https://bun.sh/blog/bun-v1.0.4)**

## What's Changed
* fix(runtime): exclude unevaluated module in `require.cache` by @Hanaasagi in https://github.com/oven-sh/bun/pull/5903
* docs: use the correct channel name for feedback by @jumoog in https://github.com/oven-sh/bun/pull/5899
* Remove bun dev stuff from create_command by @colinhacks in https://github.com/oven-sh/bun/pull/5939
* [install] fix GitHub dependency bugs by @dylan-conway in https://github.com/oven-sh/bun/pull/5941
* isBinary by @dylan-conway in https://github.com/oven-sh/bun/pull/5944
* Update installation.md by @sonyarianto in https://github.com/oven-sh/bun/pull/5968
* docs: update link to templates in remix guide by @meck93 in https://github.com/oven-sh/bun/pull/5965
* fix(lockfile): ensure all bytes of union are initialized before serialization. by @Hanaasagi in https://github.com/oven-sh/bun/pull/5957
* Fixes #5985 by @Jarred-Sumner in https://github.com/oven-sh/bun/pull/5986
* Use c-ares function for checking if a string is an IP address by @Jarred-Sumner in https://github.com/oven-sh/bun/pull/5950
* docs: Fixes link to vitejs by @aszenz in https://github.com/oven-sh/bun/pull/5959
* docs: `bun init` is missing in README by @cyfung1031 in https://github.com/oven-sh/bun/pull/5989
* Update writing.md - Remove duplicate code snippet by @jeroenvanrensen in https://github.com/oven-sh/bun/pull/5966
* Correctly fix #5888 by @Jarred-Sumner in https://github.com/oven-sh/bun/pull/6000
* Do not use removefileat() by @Jarred-Sumner in https://github.com/oven-sh/bun/pull/6001
* fix latest dev build panic by @Hanaasagi in https://github.com/oven-sh/bun/pull/6026
* Fix create command with template prefixed with @ char #6007 by @axlEscalada in https://github.com/oven-sh/bun/pull/6013
* Add fs.statfs{Sync} to missing fs apis by @techvlad in https://github.com/oven-sh/bun/pull/6030
* Make error message for `new URL(invalid)` better by @Jarred-Sumner in https://github.com/oven-sh/bun/pull/6032
* Add `Module._extensions` by @Jarred-Sumner in https://github.com/oven-sh/bun/pull/5998
* Drain microtasks at end of abort() if called into JS by @Jarred-Sumner in https://github.com/oven-sh/bun/pull/6036
* fix `captureStackTrace` inside constructor without `super` in extended by @dylan-conway in https://github.com/oven-sh/bun/pull/6063
* docs: fixed small error by @0xflotus in https://github.com/oven-sh/bun/pull/6077
* docs: update astro.md to v3 by @Dawntraoz in https://github.com/oven-sh/bun/pull/6070
* Improve Docker images by @Electroid in https://github.com/oven-sh/bun/pull/5771
* fix: Docker - Include `bunx` symlink in distroless variant by @polarathene in https://github.com/oven-sh/bun/pull/6090
* fix(fetch/server) fix server end of stream, fix fetch not streaming without content-length or chunked encoding, fix case when stream do not return a promise on pull by @cirospaciari in https://github.com/oven-sh/bun/pull/6086
* fix: support console.dir options object correctly by @liz3 in https://github.com/oven-sh/bun/pull/6059
* fix workspace dependency install by @dylan-conway in https://github.com/oven-sh/bun/pull/6092
* fix(node:fs): fix `fs.exists` callback parameters by @Hanaasagi in https://github.com/oven-sh/bun/pull/6097
* fix: Docker - Apply workaround with `RUN` to symlink `bunx` by @polarathene in https://github.com/oven-sh/bun/pull/6100
* fix: implement correct behaviour for urls with blob: scheme by @liz3 in https://github.com/oven-sh/bun/pull/5825
* Update developer arch install steps for llvm and clang by @booniepepper in https://github.com/oven-sh/bun/pull/6120
* Update run.md by @DarthDanAmesh in https://github.com/oven-sh/bun/pull/6099
* fix(bun install): Handle vercel and github tarball path dependencies by @booniepepper in https://github.com/oven-sh/bun/pull/6122
* revert fix for passing empty env vars to `bun run` by @dylan-conway in https://github.com/oven-sh/bun/pull/6123
* `deadCodeElimination` toggle for Bun.Transpiler by @jhmaster2000 in https://github.com/oven-sh/bun/pull/5932
* fix typescript metadata for import identifiers by @dylan-conway in https://github.com/oven-sh/bun/pull/6130
* Complete rework of the majority of `node:util`, primarily `util.inspect` by @jhmaster2000 in https://github.com/oven-sh/bun/pull/4493
* Get Next.js Pages Router to work by @paperdave in https://github.com/oven-sh/bun/pull/6095
* docs: update image astro.md by @Dawntraoz in https://github.com/oven-sh/bun/pull/6149
* Reduce memory usage of HTTP server by @Jarred-Sumner in https://github.com/oven-sh/bun/pull/6135
* Add local tarball install #5812 by @axlEscalada in https://github.com/oven-sh/bun/pull/6118
* docs (runtime): the order of checking files when using bare impoort by @DevinJohw in https://github.com/oven-sh/bun/pull/5919
* Upgrade to latest Node.js version by @Jarred-Sumner in https://github.com/oven-sh/bun/pull/6158
* Fixes #6053 by @Jarred-Sumner in https://github.com/oven-sh/bun/pull/6162
* feat(runtime): implement `server.requestIp` + node:http `socket.address()` by @paperdave in https://github.com/oven-sh/bun/pull/6165
* fix(resolver): support encoded file urls by @paperdave in https://github.com/oven-sh/bun/pull/5766
* fix(runtime): Socket.prototype is undefined by @paperdave in https://github.com/oven-sh/bun/pull/5945
* fix: don't set default request method when creating a Request from another by @liz3 in https://github.com/oven-sh/bun/pull/6154
* fix(runtime): followup for `server.requestIP` by @paperdave in https://github.com/oven-sh/bun/pull/6185
* Implement virtual module support in `Bun.plugin` by @Jarred-Sumner in https://github.com/oven-sh/bun/pull/6167
* Fix hang in `bun install` by @Jarred-Sumner in https://github.com/oven-sh/bun/pull/6192
* tweak github actions  by @Jarred-Sumner in https://github.com/oven-sh/bun/pull/6195
* Fix bug causing "Connection Refused" errors by @Jarred-Sumner in https://github.com/oven-sh/bun/pull/6206
* fix(node:process): fix return value of `process.kill` by @Hanaasagi in https://github.com/oven-sh/bun/pull/6207
* Slightly reduce number of open file descriptors in `bun install` by @Jarred-Sumner in https://github.com/oven-sh/bun/pull/6219
* Added the fileExtensions field to file-system-router.md by @cornedor in https://github.com/oven-sh/bun/pull/6231
* Add Bun.TOML to types by @colinhacks in https://github.com/oven-sh/bun/pull/6161
* Update development.md by @ciceropablo in https://github.com/oven-sh/bun/pull/6221
* Warn at start when using AVX build of Bun without AVX support by @Jarred-Sumner in https://github.com/oven-sh/bun/pull/6242
* Fix `bun install` reading Github API from wrong environment variable by @Electroid in https://github.com/oven-sh/bun/pull/6247
* Set `fetch` timeout to 5 minutes by @Jarred-Sumner in https://github.com/oven-sh/bun/pull/6217
* Store workspace package versions by @dylan-conway in https://github.com/oven-sh/bun/pull/6258

## New Contributors
* @meck93 made their first contribution in https://github.com/oven-sh/bun/pull/5965
* @aszenz made their first contribution in https://github.com/oven-sh/bun/pull/5959
* @cyfung1031 made their first contribution in https://github.com/oven-sh/bun/pull/5989
* @axlEscalada made their first contribution in https://github.com/oven-sh/bun/pull/6013
* @techvlad made their first contribution in https://github.com/oven-sh/bun/pull/6030
* @Dawntraoz made their first contribution in https://github.com/oven-sh/bun/pull/6070
* @polarathene made their first contribution in https://github.com/oven-sh/bun/pull/6090
* @DarthDanAmesh made their first contribution in https://github.com/oven-sh/bun/pull/6099
* @DevinJohw made their first contribution in https://github.com/oven-sh/bun/pull/5919
* @cornedor made their first contribution in https://github.com/oven-sh/bun/pull/6231
* @ciceropablo made their first contribution in https://github.com/oven-sh/bun/pull/6221

**Full Changelog**: https://github.com/oven-sh/bun/compare/bun-v1.0.3...bun-v1.0.4

<hr /><em>This discussion was created from the release <a href='https://github.com/oven-sh/bun/releases/tag/bun-v1.0.4'>Bun v1.0.4</a>.</em>

**Comment by azevedoguigo** [comment]
Installing with npm seems more user-friendly.

**Comment by abhaytomar05** [comment]
After installing bun with npm, you can replace npm with bun 🤘🏻😃

**Reply by Tuanm** [reply]
Just like how IE works years ago: for downloading Firefox. 🌞

**Comment by LucasKauz** [comment]
Awesome update, the importance of the Next.js fix is understated, it was broken without any user interaction with the default installation. I imagine a lot of people, like me, got stuck with the storage error and it was quickly fixed.

Keep it up! :1st_place_medal:
